services/horizon: Add verify-range docker image (#2106)
stellar/go: services/horizon/docker/verify-range/README.md

# `stellar/expingest-verify-range`

This docker image allows running multiple instances of `horizon expingest verify-command` on a single machine or running it in [AWS Batch](https://aws.amazon.com/batch/).

## Env variables

### Running locally

Name | Description
-|- 
`BRANCH` | Git branch to build (useful for testing PRs)
`FROM` | First ledger of the range (must be checkpoint ledger)

> `TO` | Last ledger of the range (must be checkpoint ledger)

Reviewer: Use http://markdowntable.com/ for these (note: can't auto suggest for a range)
```suggestion
| Name     | Description                                           |
|----------|-------------------------------------------------------|
| `BRANCH` | Git branch to build (useful for testing PRs)          |
| `FROM`   | First ledger of the range (must be checkpoint ledger) |
| `TO`     | Last ledger of the range (must be checkpoint ledger)  |
```

### Running in AWS Batch

Name | Description
-|- 
`BRANCH` | Git branch to build (useful for testing PRs)
`BATCH_START_LEDGER` | First ledger of the AWS Batch Job, must be a checkpoint ledger or 1.

> `BATCH_SIZE` | Size of the batch, must be multiple of 64.

Reviewer: ```suggestion
| Name                 | Description                                                          |
|----------------------|----------------------------------------------------------------------|
| `BRANCH`             | Git branch to build (useful for testing PRs)                         |
| `BATCH_START_LEDGER` | First ledger of the AWS Batch Job, must be a checkpoint ledger or 1. |
| `BATCH_SIZE`         | Size of the batch, must be multiple of 64.                           |
```

#### Example

When you start 10 jobs with `BATCH_START_LEDGER=63` and `BATCH_SIZE=64`
it will run the following ranges:

`AWS_BATCH_JOB_ARRAY_INDEX` | `FROM` | `TO`
-|-|-
0 | 63 | 127
1 | 127 | 191
2 | 191 | 255

> 3 | 255 | 319

Reviewer: ```suggestion
| `AWS_BATCH_JOB_ARRAY_INDEX` | `FROM` | `TO` |
|-----------------------------|--------|------|
| 0                           | 63     | 127  |
| 1                           | 127    | 191  |
| 2                           | 191    | 255  |
| 3                           | 255    | 319  |
```

## Tips when using AWS Batch

> * In "Job definition" set vCPUs to 2 and Memory to 4096 what represents `c5.large` instances Horizon should be using.

Reviewer: ```suggestion
* In "Job definition" set vCPUs to 2 and Memory to 4096. This represents the `c5.large` instances Horizon should be using.
```
